indico/modules/attachments/controllers/event_package.py

```diff
@@ -41,6 +41,29 @@
 from MaKaC.conference import SubContribution
 
 
+def adjust_path_length(segments):
+    """
+    Shorten the path length to < 260 chars.
+
+    Windows' built-in ZIP tool doesn't like files whose
+    total path exceeds 260 chars. Here we progressively
+    shorten the total until it matches that constraint.
+    """
+    result = []
+    total_len = sum(len(seg) for seg in segments)
+    excedent = (total_len - 255) if total_len > 255 else 0
+
+    for seg in segments[::-1]:
+        fname, ext = os.path.splitext(seg)
+        cut = min(excedent, (len(fname) - 10) if len(fname) > 14 else 0)
+        if cut:
+            excedent -= cut
+            fname = fname[:-cut]
+        result.append((fname + ext) if ext else fname)
```

Comment:
`ext` is an empty string if there's no extension, so you can always use `fname + ext`